rasa/shared/exceptions.py

```diff
@@ -13,18 +13,46 @@ class RasaXTermsError(RasaException):
     """Error in case the user didn't accept the Rasa X terms."""
 
 
-class YamlSyntaxException(RasaException):
+class YamlException(RasaException):
+    """Raised if there is an error reading yaml."""
+
+    def __init__(self, filename: Optional[Text] = None):
+        """Create exception.
+
+        Args:
+            filename: optional file the error occurred in"""
+        self.filename = filename
+
+
+class YamlSyntaxException(YamlException):
     """Raised when a YAML file can not be parsed properly due to a syntax error."""
 
-    def __init__(self, filename: Text, underlying_yaml_exception: Exception):
+    def __init__(
+        self,
+        filename: Optional[Text] = None,
+        underlying_yaml_exception: Optional[Exception] = None,
+    ):
         self.underlying_yaml_exception = underlying_yaml_exception
-        self.filename = filename
+        super(YamlSyntaxException, self).__init__(filename)
```

Comment:
optional: if possible call the `super()` thing before doing anything else https://www.notion.so/rasa/Python-Code-Conventions-26835ffd72484dd699f28fa693916d2e#498bb338d9ca4de2892a70d0e7cb71d2